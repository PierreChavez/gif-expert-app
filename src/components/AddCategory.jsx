import { useState } from "react"

export const AddCategory = ({onSetCategories}) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChange = ({ target }) => {
        setInputValue(target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault()
        if( inputValue.trim().length <= 1) return;
        onSetCategories(inputValue.trim())
        setInputValue('')
    }

    return (
        <form onSubmit={ onSubmit }>
            <input type="text"
                placeholder="Buscar Gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}
