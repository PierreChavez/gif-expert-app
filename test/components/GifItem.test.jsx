import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem"

describe('Prueba de componente GifItem', () => {

    const title = 'Prueba 999';
    const url = 'https://miurl/gitf.jpgu';

    test('debe hacer match con el snapshot', () => {

        const { container } = render(<GifItem title={title}  url={url} />)
        expect(container).toMatchSnapshot();
    })

    test('debe de mostar la imagen con el url dado y el alt indicado', () => { 

        render(<GifItem title={title}  url={url} />)

        const {src, alt} = screen.getByRole('img');

        expect(src).toBe(url);
        expect(alt).toBe(title);

    //    expect(screen.getByRole('img').alt).toBe(title)
    //    expect(screen.getByRole('img').src).toBe(url);


     })


     test('debe de mostar el titulo en el componente', () => { 

        render(<GifItem title={title}  url={url} />)

        expect(screen.getByText(title)).toBeTruthy();
        
      })

})