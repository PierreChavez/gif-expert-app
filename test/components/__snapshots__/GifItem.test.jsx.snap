// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de componente GifItem debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <p>
      Prueba 999
    </p>
    <img
      alt="Prueba 999"
      src="https://miurl/gitf.jpgu"
    />
  </div>
</div>
`;
